db/import.go

```diff
@@ -83,8 +85,17 @@ func (db *Database) importDoc(docid string, body Body, isDelete bool, existingDo
 			// If this is an on-demand import, we want to continue to import the current version of the doc.  Re-initialize existing doc based on the latest doc
 			if mode == ImportOnDemand {
 				body = doc.Body()
+
+				// Reload the doc expiry
+				gocbBucket := db.Bucket.(*base.CouchbaseBucketGoCB)
+				expiry, getExpiryErr := gocbBucket.GetExpiry(docid)
+				if getExpiryErr != nil {
+					return nil, nil, getExpiryErr
+				}
+
```

Comment:
We don't need to check if the cas here matches the doc.Cas, because if we've got a stale value in doc.Cas it's going to trigger another CAS retry anyway?

Reply:
From chat:

> initially my thought was "we need to retrieve the Cas that's associated with the expiry when we do GetExpiry, and make sure it matches doc.Cas".
> but thinking further I don't think that's actually required.  If the doc was updated in between us getting existingDoc, and getting the expiry, the subsequent write is going to fail anyway (due to CAS)

So no action needed on this